### ISSUE TITLE
How about creating a "Qt Studio" IDE

### ISSUE THREAD

**Issue by StevenZack** [therecipe/qt#373]
This project is awesome ,
I think it's possible to make this project  more popular if we can create an easy-to-use IDE like Android-Studio or QtCreator ...

I mean look at QtCreator , we can use it immediately after we install it.
It will be better if there is a integrated package for installation , which can run immediately .

Providing a installation package which include Qt5.8 , Go and Liteide(or something else) ,basically everything you need  when you wanna develop a desktop GUI app at a clean PC .
(I think the uitools part should be a standard GUI app template , because it separate logic part and UI part .Just like Android Studio and Qt Creator)
We can let the Qt Creator handle the UI part , let Liteide ( or other go IDE) handle the logic part .

It's just a thought .I will help if I can .

**Comment by therecipe** [comment]
Hey

Thanks for the suggestion and also for offer to help, I really apprechiate it :)
I really the the idea of bundling all this inside a single setup, but I fear it's a bit out of scope for now.

The only way I see to make it easily work for now, would be to modfiy Qt's online installer. 
It's possible: http://doc.qt.io/qtinstallerframework/index.html, but I'm not sure if there are any legal obligations.

**Comment by hubyhuby** [comment]
Yes, just to confirm that would help a lot new bees like me :)
I d like to do a Prof of concept app for viewing photos.
But just starting the right seems complicated ;)
Bravo for this wonderful work on QT !

**Comment by amlwwalker** [comment]
On the back of this project, I am working towards that - its a github repo so obviously not install and go but once setup is done its easy to develop android/windows/osx/linux apps because there is also a hotloader, so when the app is running, any changes to the qml auto reload the UI. If you're _reasonably_ able to do the setup, I find it a nice way to build apps: https://github.com/amlwwalker/got-qt

**Comment by therecipe** [comment]
Looked into this a while ago: https://github.com/therecipe/qt/issues/565#issuecomment-376087787
And yeah, it's possible to provide prebuilt libs and bundle this whole package inside a single installation. But it will take some time, because there is a lot of stuff needed to be done first to get this working properly.
There is also the option to get this working quick and dirty, but that's probably to intensive to maintain.
However, I will try to get this going in the next 3 month.

**Comment by ManPython** [comment]
@StevenZack 

I'm not sure about Your post.

1. Is this way in direction that we have some GoIDE similar live MS Visual, Embarcadero, Lazarus, SharpDevelop? 
2. Based on QT?
3. If yes, than great..
a) Maybe this need some Kickstarter and support

4. Here is problem
a) GOlang replicated python pip/apt based on github
b) GOlang is related with Google
c) MS buy github  (tak care a), b) upper), this can be punch for Google and GOlang (both MS/G are devils : )
d) Google don't care desktop, the live with on-line aps and web services and most business model is in this direction (cloud, remote, etc.. controled data) 
- the quality off apps going down and was simplified radicaly
e) MS creating c# as modern lang, but they producing StarTrek Enterprise wehicle (some problem about backward compatibility)

5. Soo.. this way is good for GOlang, but this lang is young and problematic in my opinion in something big as IDE, but has good potential.

6. I belive that we still wasting time around configure and prepare coding enviorment more than real coding
a) we meeting to much additionl libs, solutions
b) the solutions has added to much technologies
..it's sometimes bether to open beer and watching how they doing this and smoke some cigaretos..  : )
c) soo.. thats way born nodeJS where everything want be in 1 lang ~~1technology and 1000 frameworks : )

7. Can be GOlang and QT some "RAPID" option? Sure.. 
a) but who can do this?
b) how much cost?

8. I'm pretty sure that QT GUI is good standard and still good for developing aps not like MS/VS

9. But I'm not sure about GOlang in this way

10. In case of any sucess of work with GOlang therecipe/qt ..
a) how many aps are developed in this bundle as sucess or commercial?
b) maybe we need some pools here (is this easy to install, configure, prepare, build, or is this good way)
c) in other (negative scenario) this can be like with pint 6.c) ..they do this with JS or similray Adobe Flash/Air, but not predicted and not developed for real advanced of any stable app that need something more to do than show some tableview (this is case of use python and PyQt, with rescue and panic run for some stron executable way for exe like "niutka", etc)
d) soo.. bad MS.. but they doing this job in some aceptable level
..we are alone here ;]  ..no cordination..but think all about.. it's to compare with Win (install and do job if working) or Lnx (install, configure, search, read, check, search, read, check.. spent some time.. and try not broke anything in next steps, coz you can be in trouble and again ..search.. or reinstall)
e) Soo.. the idea of IDE can solve problem of "complexity" of multiple tech/sol. etc.. if is possible and not only "install all what is need" producing 20GB of stuff that can be easy to broke by some other stuff or configuration

**Comment by Petricevic** [comment]
The project is really fantastic, and even the idea of an IDE is really good. I have experience with GOlang but I'm a beginner with android and Qt and installing the ambian in debian 9 for me is really complex. It always keeps missing something and I can not fill out the examples. Many years ago PowerBuilder existed as a wonderful cross-platform IDE, with ver. 4.0 had become a C code generator. It still exists, APPEON sells it but has been useless for years. I think the idea of a multiplatform graphic environment like Delphy / Lazarus with a very skeletal macrolanguage like PowerBuilder that generates GOlang code with Qt for the graphic part and objects dedicated to dialogue with the database as PowerBuilder would be a big success, maybe even commercial . If someone decided to take it forward I would be happy to participate with my modest skills.
